### DISCUSSION TITLE
[Bug]: Got error but the same prompt and prefix worked in Openai playground

### DISCUSSION THREAD

**Discussion by Niqnil** [danny-avila/LibreChat#1642] [Troubleshooting] [answered]
### What happened?

I was trying to use Openai's gpt-4-turbo-preview on LibreChat v0.6.6 on Brave browser and got this error message:
"Something went wrong. Here's the specific error message we encountered: Prompt token count of 62652 exceeds max token count of 8187."

I tried the exact same prompt in Openai playground and it worked. I cannot post the prompt because it needs to be kept private.

I'm not sure where token count of 8187 is coming from as it does not seem to match any of Openai's models. Could it be related to my setup to use OpenRouter?

```
# my librechat.yaml

# Configuration version (required)
version: 1.0.1

# Cache settings: Set to true to enable caching
cache: true

# Definition of custom endpoints
endpoints:
  custom:
    # Mistral AI API
    - name: "Mistral"  # Unique name for the endpoint
      # For `apiKey` and `baseURL`, you can use environment variables that you define.
      # recommended environment variables:
      apiKey: "${MISTRAL_API_KEY}"
      baseURL: "https://api.mistral.ai/v1"

      # Models configuration
      models: 
        # List of default models to use. At least one value is required.
        default: ["mistral-tiny", "mistral-small", "mistral-medium"]
        # Fetch option: Set to true to fetch models from API.
        fetch: true  # Defaults to false.

      # Optional configurations
      
      # Title Conversation setting
      titleConvo: true  # Set to true to enable title conversation

      # Title Method: Choose between "completion" or "functions".
      titleMethod: "completion"  # Defaults to "completion" if omitted.

      # Title Model: Specify the model to use for titles.
      titleModel: "mistral-tiny"  # Defaults to "gpt-3.5-turbo" if omitted.

      # Summarize setting: Set to true to enable summarization.
      summarize: false

      # Summary Model: Specify the model to use if summarization is enabled.
      summaryModel: "mistral-tiny"  # Defaults to "gpt-3.5-turbo" if omitted.

      # Force Prompt setting: If true, sends a `prompt` parameter instead of `messages`.
      forcePrompt: false

      # The label displayed for the AI model in messages.
      modelDisplayLabel: "Mistral"  # Default is "AI" when not set.

      # Add additional parameters to the request. Default params will be overwritten.
      addParams:
        safe_prompt: true # This field is specific to Mistral AI: https://docs.mistral.ai/api/
        
      # Drop Default params parameters from the request. See default params in guide linked below.
      # NOTE: For Mistral, it is necessary to drop the following parameters or you will encounter a 422 Error:
      dropParams: ["stop", "user", "frequency_penalty", "presence_penalty"]
      
    # OpenRouter.ai Example
    - name: "OpenRouter"
      # For `apiKey` and `baseURL`, you can use environment variables that you define.
      # recommended environment variables:
      # Known issue: you should not use `OPENROUTER_API_KEY` as it will then override the `openAI` endpoint to use OpenRouter as well.
      apiKey: "${OPENROUTER_KEY}"
      baseURL: "https://openrouter.ai/api/v1"
      models:
        default: ["gpt-3.5-turbo"]
        fetch: true
      titleConvo: true
      titleModel: "mixtral-8x7b-instruct"
      summarize: false
      summaryModel: "gpt-3.5-turbo"
      forcePrompt: false
      modelDisplayLabel: "OpenRouter"

# See the Custom Configuration Guide for more information:
# https://docs.librechat.ai/install/configuration/custom_config.html
```
```
# my .env

OPENAI_API_KEY=REDACTED
# OPENAI_MODELS=gpt-3.5-turbo-1106,gpt-4-1106-preview,gpt-3.5-turbo,gpt-3.5-turbo-16k,gpt-3.5-turbo-0301,gpt-4,gpt-4-0314,gpt-4-0613

DEBUG_OPENAI=false

# TITLE_CONVO=false
# OPENAI_TITLE_MODEL=gpt-3.5-turbo

# OPENAI_SUMMARIZE=true
# OPENAI_SUMMARY_MODEL=gpt-3.5-turbo

# OPENAI_FORCE_PROMPT=true

# OPENAI_REVERSE_PROXY=

# OPENAI_ORGANIZATION= 

OPENROUTER_KEY=REDACTED
```

### Steps to Reproduce

NA

### What browsers are you seeing the problem on?

Chrome

### Screenshots

[https://i.ibb.co/9pYjfM3/image.png](url)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
The `gpt-4-turbo-preview` alias was just released, I'll be adding it now

**Reply by Niqnil** [reply]
By the way, same error happens if gpt-4-0125-preview is selected.

**Reply by danny-avila** [reply]
Thanks I had this noted as well. I read the announcement but didn't realize it was for today, d'oh

**Reply by danny-avila** [reply]
Merging soon! https://github.com/danny-avila/LibreChat/pull/1643

**Reply by Niqnil** [reply]
Just saw a similar error while trying OpenRouter anthropic/claude-2 for the first time. "Something went wrong. Here's the specific error message we encountered: Prompt token count of 62652 exceeds max token count of 4095."
[Screenshot](https://i.ibb.co/QmnPVnM/image.png)

**Reply by Passerby1011** [reply]
Same problem, I added a custom endpoint and added the parameter max_tokens, but it didn't seem to take effect.
![image](https://github.com/danny-avila/LibreChat/assets/63802758/9e6caba5-959b-4a3c-8a27-25de69ff9e1e)
![image](https://github.com/danny-avila/LibreChat/assets/63802758/156c7153-5b15-43f0-af7e-132aae036b8a)

**Reply by danny-avila** [reply]
I will add the anthropic models to the list of recognized models for the custom endpoint

> OpenRouter anthropic/claude-2 for the first time

Openrouter is unique in and of itself, since we can actually fetch what the max context is and even the rate they charge. I have been thinking of how to sync up with their data for a more robust approach

**Reply by danny-avila** [reply]
A fix for this is merged

**Reply by Niqnil** [reply]
OpenRouter anthropic/claude-2  works now. Thank you:)

Saw this error when trying Google gemini-pro. 
"Something went wrong. Here's the specific error message we encountered: [GoogleGenerativeAI Error]: Error fetching from https://generativelanguage.googleapis.com/v1/models/gemini-pro:streamGenerateContent?alt=sse: [500 Internal Server Error] An internal error has occurred. Please retry or report in https://developers.generativeai.google/guide/troubleshooting"

Is there a problem with this endpoint too?
```
# .env snippet
# other GOOGLE* settings were not set. Do they need to be?

GOOGLE_KEY=REDACTED
GOOGLE_MODELS=gemini-pro,gemini-pro-vision
```
[Screenshot](https://i.ibb.co/kJzG087/image.png)

**Reply by Niqnil** [reply]
Can you add Openrouter's codellama/codellama-70b-instruct too? It currently shows error "Something went wrong. Here's the specific error message we encountered: Prompt token count of 30799 exceeds max token count of 4095." when I try it.

**Reply by danny-avila** [reply, marked as answer]
the google error is specific to Google's servers, not sure why, it has nothing to do with the app.

> Can you add Openrouter's codellama/codellama-70b-instruct too? It currently shows error "Something went wrong. Here's the specific error message we encountered: Prompt token count of 30799 exceeds max token count of 4095." when I try it.

Sure, I'll have a solution for all Openrouter models in general, soon.

**Reply by danny-avila** [reply]
@Niqnil this is addressed here: https://github.com/danny-avila/LibreChat/pull/1703
